### PR TITLE
CX chains wallets

### PR DESCRIPTION
**Pull request by amherag**
Changes:
- Added capability of creating a wallet through --create-wallet and -wallet-seed

Does this change need to mentioned in CHANGELOG.md?
Yes.

### CODE REVIEW

File: cxgo/main.go [hunk 1]
```diff
@@ -300,6 +301,53 @@ func runNode (mode string, options cxCmdFlags) *exec.Cmd {
 	}
 }
 
+func createWallet(csrfToken string, port int, seed string) {
+	sURL := fmt.Sprintf("http://127.0.0.1:%d/api/v1/wallet/create", port + 420)
+
+	data := url.Values{}
+	data.Set("seed", seed)
+	data.Add("coin", "cxcoin")
+	data.Add("label", "cxcoin")
```
Comment (ingwal) [inline cxgo/main.go]: Should these two really have the same value or is it a cut-and-paste bug?

Comment (amherag) [reply cxgo/main.go]: As far as I can tell, it seems that "label" needs to match the one set in the `cmd/cxcoin` executable that is generated by the `newcoin` command. By default, CX generates the executable using the "cxcoin" label too. I'm not sure if it's a good idea for CX chains that users have the capability of generating different "coins".

Regarding the "coin" key, it seems that it is non-existent, according to Skycoin's REST API. I'm removing it. The `wallet/create` endpoint does generate a JSON with a "coin" key, but it doesn't accept that parameter.

Comment (amherag) [reply cxgo/main.go]: It is removed now.

Comment (gz-c) [reply cxgo/main.go]: The `"coin"` parameter inside of the wallet JSON file is either "skycoin" or "bitcoin" to refer to the address types held in the wallet.  Allow it to default to "skycoin"

Comment (gz-c) [reply cxgo/main.go]: `"label"` is what the user would see if the file was loaded into the desktop wallet software

Comment (gz-c) [reply cxgo/main.go]: By the way you can use https://godoc.org/github.com/skycoin/skycoin/src/api#Client.CreateUnencryptedWallet and delete all of these functions

Comment (amherag) [reply cxgo/main.go]: > The `"coin"` parameter inside of the wallet JSON file is either "skycoin" or "bitcoin" to refer to the address types held in the wallet. Allow it to default to "skycoin"

Ah, I see! Thank you for the explanation.


> `"label"` is what the user would see if the file was loaded into the desktop wallet software

Then I think it is safe for now to set it to a fixed value. I'm not sure if we are ever going to create something similar to the desktop wallet for CX chains.



> 
> 
> You can use https://godoc.org/github.com/skycoin/skycoin/src/api#Client.CreateUnencryptedWallet and delete all of these functions

Excellent. Thank you for the suggestion. I never paid too much attention to the `api` package when making the necessary modifications to the Skycoin source code in order to run CX chains on it. Now I notice there are many things in `cxgo/main.go` that can be dropped in favor to using functions from this package. These modifications should be part of another PR though. Thanks again.

**Pull request merged**